apps/client/src/app/pages/inventory-optimizer/inventory-optimizer/inventory-optimizer.component.ts

```diff
@@ -120,14 +120,41 @@ export class InventoryOptimizerComponent {
   }
 
   constructor(private inventoryFacade: InventoryFacade,
-              @Inject(INVENTORY_OPTIMIZER) private optimizers: InventoryOptimizer[],
-              private lazyData: LazyDataService, private message: NzMessageService, private translate: TranslateService) {
+    @Inject(INVENTORY_OPTIMIZER) private optimizers: InventoryOptimizer[],
+    private lazyData: LazyDataService, private message: NzMessageService, private translate: TranslateService) {
   }
 
   private getContainerName(item: InventoryItem): string {
     return item.retainerName || this.inventoryFacade.getContainerName(item.containerId);
   }
 
+  public getExpansions(): string[] {
+    let expansions: any[] = this.lazyData.patches.map(p => {
+      return { name: p.ExName, version: p.ExVersion }
+    });
+
+    return expansions.filter((value, index, array) => {
+      return index === array.findIndex(e => e.name == value.name);
+    });
```

Comment:
Could probably be changed for a `uniqBy(expansions, 'name');` call, using `uniqBy` from lodash.

Reply:
What's the benefit? I'm not familiar with it.

Reply:
Lodash provides small optimizations on various methods and that's why when they have an utility method that we can use, it's better to just use it instead of implementing our own solution.

Reply:
Makes sense. The change will be in my next push.